binr/rabin2/rabin2.c

```diff
@@ -1,1120 +1,8 @@
-	if (isradjson) {
-		r_cons_print ("{");
-	}
-	// List fatmach0 sub-binaries, etc
-	if (action & R_BIN_REQ_LISTARCHS || ((arch || bits || arch_name) &&
-		!r_bin_select (bin, arch, bits, arch_name))) {
-		if (rad == R_CORE_BIN_SIMPLEST || rad == R_CORE_BIN_SIMPLE) {
-			r_bin_list_archs (bin, 'q');
-		} else {
-			r_bin_list_archs (bin, (rad == R_CORE_BIN_JSON)? 'j': 1);
-		}
-		actions_done++;
-		free (arch_name);
-	}
-	if (action & R_BIN_REQ_PDB_DWNLD) {
-		SPDBOptions pdbopts;
-		pdbopts.user_agent = (char*) r_config_get (core.config, "pdb.useragent");
-		pdbopts.symbol_server = (char*) r_config_get (core.config, "pdb.server");
-		pdbopts.extract = r_config_get_i (core.config, "pdb.extract");
-
-		if ((tmp = r_sys_getenv ("RABIN2_SYMSTORE"))) {
-			r_config_set (core.config, "pdb.symstore", tmp);
-			R_FREE (tmp);
-		}
-		pdbopts.symbol_store_path = (char*) r_config_get (core.config, "pdb.symstore");
-		int r = r_bin_pdb_download (&core, isradjson, &actions_done, &pdbopts);
-		r_core_fini (&core);
-		return r;
-	}
-
-	if ((tmp = r_sys_getenv ("RABIN2_PREFIX"))) {
-		r_config_set (core.config, "bin.prefix", tmp);
-		free (tmp);
-	}
-
-	run_action ("sections", R_BIN_REQ_SECTIONS, R_CORE_BIN_ACC_SECTIONS);
-	run_action ("entries", R_BIN_REQ_ENTRIES, R_CORE_BIN_ACC_ENTRIES);
-	run_action ("entries", R_BIN_REQ_INITFINI, R_CORE_BIN_ACC_INITFINI);
-	run_action ("main", R_BIN_REQ_MAIN, R_CORE_BIN_ACC_MAIN);
-	run_action ("imports", R_BIN_REQ_IMPORTS, R_CORE_BIN_ACC_IMPORTS);
-	run_action ("classes", R_BIN_REQ_CLASSES, R_CORE_BIN_ACC_CLASSES);
-	run_action ("symbols", R_BIN_REQ_SYMBOLS, R_CORE_BIN_ACC_SYMBOLS);
-	run_action ("exports", R_BIN_REQ_EXPORTS, R_CORE_BIN_ACC_EXPORTS);
-	run_action ("resources", R_BIN_REQ_RESOURCES, R_CORE_BIN_ACC_RESOURCES);
-	run_action ("strings", R_BIN_REQ_STRINGS, R_CORE_BIN_ACC_STRINGS);
-	run_action ("info", R_BIN_REQ_INFO, R_CORE_BIN_ACC_INFO);
-	run_action ("fields", R_BIN_REQ_FIELDS, R_CORE_BIN_ACC_FIELDS);
-	run_action ("header", R_BIN_REQ_HEADER, R_CORE_BIN_ACC_HEADER);
-	run_action ("libs", R_BIN_REQ_LIBS, R_CORE_BIN_ACC_LIBS);
-	run_action ("relocs", R_BIN_REQ_RELOCS, R_CORE_BIN_ACC_RELOCS);
-	run_action ("dwarf", R_BIN_REQ_DWARF, R_CORE_BIN_ACC_DWARF);
-	run_action ("pdb", R_BIN_REQ_PDB, R_CORE_BIN_ACC_PDB);
-	run_action ("size", R_BIN_REQ_SIZE, R_CORE_BIN_ACC_SIZE);
-	run_action ("versioninfo", R_BIN_REQ_VERSIONINFO, R_CORE_BIN_ACC_VERSIONINFO);
-	if (action & R_BIN_REQ_SRCLINE) {
-		rabin_show_srcline (at);
-	}
-	if (action & R_BIN_REQ_EXTRACT) {
-		RBinFile *bf = r_bin_cur (bin);
-		if (bf && bf->xtr_data) {
-			rabin_extract ((!arch && !arch_name && !bits));
-		} else {
-			eprintf (
-				"Cannot extract bins from '%s'. No supported "
-				"plugins found!\n",
-				bin->file);
-		}
-	}
-	if (op && action & R_BIN_REQ_OPERATION) {
-		rabin_do_operation (op);
-	}
-	if (isradjson) {
-		printf ("}");
-	}
-	r_cons_flush ();
-	r_core_fini (&core);
-	free (stdin_buf);
 
-	return 0;
+	return rabin_main (argc, argv);
```

Comment:
Maybe we should use r_main as namespace prefix.

`return r_main_rabin2 (argc, argv);`